agent/grpc-external/services/peerstream/stream_test.go

```diff
@@ -1159,7 +1159,7 @@ func TestStreamResources_Server_CARootUpdates(t *testing.T) {
 
 func TestStreamResources_Server_AckNackNonce(t *testing.T) {
 	srv, store := newTestServer(t, func(c *Config) {
-		c.incomingHeartbeatTimeout = 10 * time.Millisecond
+		c.incomingHeartbeatTimeout = 500 * time.Millisecond
```

Comment:
Thanks for asking. I think the change has minimal impact on the test flakiness if we set the `incomingHeartbeatTimeout` properly. That is, since the sender sends a message at `(time.Second / time.Duration(defaultOutgoingStreamSendLimit)` tick (or in 100ms tick), we need to give more time to the heartbeat process to wait for a message.

To clarify the change, instead of hardcoding the heartbeattimeout, I changed everything to in the updated PR.

```
// Give handler enough time to send an outgoing message before heartbeat timeout
c.incomingHeartbeatTimeout = 5 * (time.Second / time.Duration(defaultOutgoingStreamSendLimit))
```